src/gui/attributetable/qgsdualview.cpp

```diff
@@ -85,6 +85,8 @@ void QgsDualView::init( QgsVectorLayer* layer, QgsMapCanvas* mapCanvas, const Qg
   else
     columnBoxInit();
 
+  mTableView->resizeColumnsToContents();
```

Comment:
This kills the performance of the attribute table a lot. The table of 5000 records normally opens right away but now takes ages to open and uses heaps of memory.